.github/workflows/test.yml

```diff
@@ -3,45 +3,84 @@ name: Run DTS tests
 on:
   pull_request:
     branches:
-      - 'main'
+      - "main"
 jobs:
   build-dts:
     uses: ./.github/workflows/build.yml
     with:
       cacheless: false
+
   run-tests:
     name: Run DTS tests
     needs: build-dts
     runs-on:
       labels: dts-builder
+    strategy:
+      matrix:
+        include:
+          - test: msi_comm
+            dpp_download_key: $MSI_DOWNLOAD
+            dpp_logs_key: $MSI_LOGS
+            log_file: output_msi.log
+
+          - test: msi_dpp
+            dpp_download_key: $MSI_DOWNLOAD
+            dpp_logs_key: $MSI_LOGS
+            log_file: output_msi.log
+
+          - test: msi_heads
+            dpp_download_key: $MSI_HEADS_DOWNLOAD
+            dpp_logs_key: $MSI_HEADS_LOGS
+            log_file: output_msi_heads.log
+
+          - test: optiplex_dpp
+            dpp_download_key: $OPTIPLEX_DOWNLOAD
+            dpp_logs_key: $OPTIPLEX_LOGS
+            log_file: output_nc_heads.log
+
+          - test: novacustom_heads
+            dpp_download_key: $OPTIPLEX_DOWNLOAD
+            dpp_logs_key: $OPTIPLEX_LOGS
+            log_file: output_nc_heads.log
+
+          - test: novacustom_comm
+            dpp_download_key: $OPTIPLEX_DOWNLOAD
+            dpp_logs_key: $OPTIPLEX_LOGS
+            log_file: output_nc_heads.log
+
+          - test: pcengines_dpp
+            dpp_download_key: $PCENGINES_DOWNLOAD
+            dpp_logs_key: $PCENGINES_LOGS
+            log_file: output_pcengines.log
+
+          - test: pcengines_seabios
+            dpp_download_key: $PCENGINES_SEABIOS_DOWNLOAD
+            dpp_logs_key: $PCENGINES_SEABIOS_LOGS
+            log_file: output_pcengines_seabios.log
+
     steps:
       - name: Checkout OSFV repo
         uses: actions/checkout@v4
         with:
-          repository: 'Dasharo/open-source-firmware-validation'
-          path: 'open-source-firmware-validation'
-          submodules: 'recursive'
-          ref: 'develop'
+          repository: "Dasharo/open-source-firmware-validation"
```

Comment:
You can checkout this repo only once in earlier job as I'm not sure if it'll succeed if folder already exist.

Reply:
Fixed: 44075c2eec038732548a90254a3a435ba1be8173